.github/detect-secrets.yaml

```diff
@@ -0,0 +1,65 @@
+name: "Secret Detection"
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+
+jobs:
+  secret-detection:
+    name: Secret-Detection
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
+      contents: read
+      security-events: write
```

Comment:
Do we use these write permissions in the workflow? It doesn't seem to be used here

Reply:
I'm not super familiar with how permissions work for github actions, is write permission needed if any changes are made to the local file system, for example `cp -pr .secrets.baseline .secrets.new`?

Reply:
No, filesystem stuff is enabled by default; so should read permissions for the repositories. We should be good to remove that permissions block